spec/services/invoice_renderer_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'spec_helper'
+
+describe InvoiceRenderer do
+  let(:service) { described_class.new }
+
+  it "creates a PDF invoice with two different templates" do
+    order = create(:completed_order_with_fees)
+    order.bill_address = order.ship_address
+    order.save!
+
+    result = service.render_to_string(order)
+    expect(result).to match /^%PDF/
+
+    allow(Spree::Config).to receive(:invoice_style2?).and_return true
+
+    alternative = service.render_to_string(order)
+    expect(alternative).to match /^%PDF/
+    expect(alternative).to_not eq result
+  end
```

Comment:
IMO this deserves to be split in two tests, one for each `invoice_template` branch, which it'd make the class and the test more maintainable. Besides, as it is, looks more like a capybara test.

Reply:
Haha, I had a comment there explaining why and then removed it. This is the second time that I removed a comment and then you ask me about it. I should never remove one of my comments again.

Here are some reasons:

- You can't test if the two results are different without rendering both results in the same test.
- Splitting the test means generating the pdf files several times which costs spec running time.

What do you think? Is it worth it splitting the spec into three and pay with 6 seconds more run time?

1. it renders an invoice pdf
2. it renders an alternative invoice pdf
3. it renders two different pdfs

Reply:
> This is the second time that I removed a comment and then you ask me about it. I should never remove one of my comments again.

:trollface: